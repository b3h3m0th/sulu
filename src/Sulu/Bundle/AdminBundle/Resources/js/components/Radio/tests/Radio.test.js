// @flow
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Radio from '../Radio';

test('The component should render in light skin', () => {
    const {container} = render(<Radio skin="light" />);
    expect(container).toMatchSnapshot();
});

test('The component should render in dark skin', () => {
    const {container} = render(<Radio skin="dark" />);
    expect(container).toMatchSnapshot();
});

test('The component should render in disabled state', () => {
    const {container} = render(<Radio disabled={true} />);
    expect(container).toMatchSnapshot();
});

test('The component pass the props correctly to the generic checkbox', () => {
    render(
        <Radio
            checked={true}
            disabled={true}
            name="my-name"
            value="my-value"
        >
            My label
        </Radio>
    );

    const checkbox = screen.queryByDisplayValue('my-value');

    expect(checkbox.value).toEqual('my-value');
    expect(checkbox).toHaveAttribute('name', 'my-name');
    expect(screen.getByText('My label')).toBeInTheDocument();
    expect(checkbox).toBeChecked();
    expect(checkbox).toBeDisabled();
});

test('The component pass the the value to the change callback', async() => {
    const onChange = jest.fn();
    render(<Radio onChange={onChange} value="my-value">My label</Radio>);

    const checkbox = screen.queryByDisplayValue('my-value');
    await userEvent.click(checkbox);

    expect(onChange).toHaveBeenCalledWith('my-value');
});
