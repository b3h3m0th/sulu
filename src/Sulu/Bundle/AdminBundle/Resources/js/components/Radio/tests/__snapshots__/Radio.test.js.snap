// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render in dark skin 1`] = `
<div>
  <label
    class="label"
    tabindex="-1"
  >
    <span
      class="switch radio dark"
    >
      <input
        type="radio"
        value=""
      />
      <span />
    </span>
  </label>
</div>
`;

exports[`The component should render in disabled state 1`] = `
<div>
  <label
    class="label disabled"
    tabindex="-1"
  >
    <span
      class="switch disabled radio dark disabled"
    >
      <input
        disabled=""
        type="radio"
        value=""
      />
      <span />
    </span>
  </label>
</div>
`;

exports[`The component should render in light skin 1`] = `
<div>
  <label
    class="label"
    tabindex="-1"
  >
    <span
      class="switch radio light"
    >
      <input
        type="radio"
        value=""
      />
      <span />
    </span>
  </label>
</div>
`;
